pages/applications/create.tsx

```diff
@@ -0,0 +1,197 @@
+import { Nullable } from '../../types/utils';
+import { createJsonResponse } from '../../utils/http/httpHelpers';
+import { splitByWhitespaces } from '../../utils/strings/formatters';
+import moment from 'moment';
+import { roleTypeToDisplayStringMap } from '../../utils/role/roleUtils';
+
+function ApplicationCreate() {
+  const router = useRouter();
+
+  const [isCreateCompanyFormOpen, setIsCreateCompanyFormOpen] = useState<boolean>(false);
+  const [isCreateRoleFormOpen, setIsCreateRoleFormOpen] = useState<boolean>(false);
+
+  const [companySearchParams, setCompanySearchParams] = useState<CompanyQueryParams>({ companyNames: [] });
+  const [selectedCompany, setSelectedCompany] = useState<Nullable<CompanyListData>>(null);
+
+  const [roleSearchParams, setRoleSearchParams] = useState<RoleQueryParams>({ searchWords: [] });
+  const [selectedRole, setSelectedRole] = useState<Nullable<RoleData>>(null);
+  const [applicationDate, setApplicationDate] = useState<Nullable<moment.Moment>>(moment(new Date()));
+
+  const [shouldShowValidationErrors, setShouldShowValidationErrors] = useState(false);
+
+  const { data: companiesData, mutate: mutateCompaniesData } = useSWR<ApiResponse<CompanyListData[]>>(
+    [COMPANIES_API_ENDPOINT, companySearchParams],
+    (url: string, companySearchParams: CompanyQueryParams) => companiesApi.getCompanies(companySearchParams),
+  );
+  const companies = companiesData?.payload || [];
+  const companyOptions: CompanyAutocompleteOption[] = companies.map((company) => ({
+    company,
+    label: <CompanyOption company={company} />,
+    value: company.name,
+  }));
+  const companyOptionsWithAdd: CompanyAutocompleteOption[] = [
+    ...companyOptions,
+    { company: null, label: <div>Add new company</div>, value: 'Add new company' },
+  ];
+
+  const onSelectCompany = (value: string, { company }: CompanyAutocompleteOption) => {
+    setSelectedCompany(company);
+
+    // Update role search params to filter for roles of the currently selected company
+    setRoleSearchParams((prevState) => ({ ...prevState, companyId: company?.id ?? undefined }));
+
+    if (company !== selectedCompany) {
+      // Unselect role that belongs to another company
+      setSelectedRole(null);
+    }
+
+    if (company === null) {
+      setIsCreateCompanyFormOpen(true);
+    }
+  };
+
+  const onSearchCompany = (inputValue: string) => {
+    setCompanySearchParams({ companyNames: splitByWhitespaces(inputValue) });
+  };
+
+  const onCreateCompany = (company: CompanyListData) => {
+    mutateCompaniesData();
+    setSelectedCompany(company);
+  };
+
+  const { data: rolesData, mutate: mutateRolesData } = useSWR<ApiResponse<RoleListData[]>>(
+    [ROLES_API_ENDPOINT, roleSearchParams],
+    (url: string, roleSearchParams: RoleQueryParams) =>
+      roleSearchParams.companyId !== undefined ? rolesApi.getRoles(roleSearchParams) : createJsonResponse([]),
```

Comment:
We dont have to handle this here. The backend does it for us i think.

```suggestion
    (url: string, roleSearchParams: RoleQueryParams) => rolesApi.getRoles(roleSearchParams),
```